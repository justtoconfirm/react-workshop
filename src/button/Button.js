import React from 'react'
import classNames from 'classnames'

const Button = (props) => {
  const {
    type = 'button',
    url = '',
    className = '',
    theme = '',
    isFullWidth = false,
    children
  } = props

  const buttonClass = classNames('btn', className, {
    'btn--primary': theme === 'primary',
    'btn--secondary': theme === 'secondary',
    'btn--full-width': isFullWidth
  })

  const isAnchorElement = url && (url.includes('http') || url.startsWith('#') || url.startsWith('mailto') || url.startsWith('/'))

  const renderButton = () =>
    <button {...{ type }} className={buttonClass}>{children}</button>

  const renderAnchor = () =>
    <a href={url} className={buttonClass}>{children}</a>

  return (
    isAnchorElement ? renderAnchor() : renderButton()
  )
}

export default Button
