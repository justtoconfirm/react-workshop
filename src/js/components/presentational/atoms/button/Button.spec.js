import Button from './Button'

describe('<Button/>', () => {
  it('should render without throwing an error', () => {
    expect(shallow(<Button/>).find('.btn').exists()).toBe(true)
  })
})
