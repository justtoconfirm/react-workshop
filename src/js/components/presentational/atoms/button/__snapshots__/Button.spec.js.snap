// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button renders correctly 1`] = `ShallowWrapper {}`;
